web3/personal/api.go

```diff
@@ -0,0 +1,95 @@
+package personal
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/status-im/status-go/geth/account"
+	"github.com/status-im/status-go/geth/params"
+	"github.com/status-im/status-go/geth/rpc"
+	"github.com/status-im/status-go/sign"
+)
+
+// ErrInvalidPersonalSignAccount is returned when the account passed to
+// personal_sign isn't equal to the currently selecte account.
+var ErrInvalidPersonalSignAccount = errors.New("invalid account. it is only possible to generate signature for a selected account")
+
+type metadata struct {
+	Data    interface{} `json:"data"`
+	Address string      `json:"account"`
+}
+
+func newMetadata(rpcParams []interface{}) (*metadata, error) {
+	data := rpcParams[0]
```

Comment:
It's gonna throw an error if a number of params is different.

Reply:
good catch!